src/BloomExe/Book/TranslationGroupManager.cs

```diff
@@ -244,16 +244,32 @@ private static void MakeElementWithLanguageForOneGroup(XmlElement groupElement,
 		/// <param name="element"></param>
 		private static void StripOutText(XmlNode element)
 		{
-			
-
-			foreach(XmlNode node in element.SelectNodes(".//*[(self::p or self::br) and not(contains(@class,'bloom-cloneToOtherLanguages'))]"))
+			var listToRemove = new List<XmlNode>();
+			foreach (XmlNode node in element.SelectNodes("descendant-or-self::*[(self::p or self::br) and not(contains(@class,'bloom-cloneToOtherLanguages'))]"))
 			{
-				node.ParentNode.RemoveChild(node);
+				listToRemove.Add(node);
 			}
 			// clean up any remaining texts that weren't enclosed
-			foreach(XmlNode node in element.SelectNodes(".//*[not(contains(@class,'bloom-cloneToOtherLanguages'))]//text()"))
+			foreach (XmlNode node in element.SelectNodes("descendant-or-self::*[not(contains(@class,'bloom-cloneToOtherLanguages'))]/text()"))
+			{
+				listToRemove.Add(node);
+			}
+			RemoveXmlChildren(listToRemove);
+		}
+
+		private static void RemoveXmlChildren(List<XmlNode> removalList)
+		{
+			foreach (var node in removalList)
 			{
-				node.ParentNode.RemoveChild(node);
+				try
+				{
+					node.ParentNode.RemoveChild(node);
+				}
+				catch (ArgumentException)
+				{
+					// Means we already deleted an intermediate node containing this one
```

Comment:
Removed the try-catch. You're right, it shouldn't be possible to cause that exception now.